packages/cache/src/cache.ts

```diff
@@ -0,0 +1,23 @@
+import { getCachedAdapter, setCachedAdapter } from "./config";
+
+class Cache {
+  async get(key: string) {
+    const cachedAdapter = await this.#getCachedAdapter()
+
+    return cachedAdapter?.get(key);
+  }
+
+  async #getCachedAdapter() {
```

Comment:
Loved the use of #method here.

I did a comment on `src/config.ts` i think this will need some changes, but you got the idea. And i think you can specify the type of the return here or throw an error if a cache adapter instance does not exist.

create a custom Exception/Error class like
```ts
class NotSpecifiedCacheAdapterError extends Error {
    constructor() {
        super('You should specify the cache adapter before trying to use the cache');
    }
}
```

And then you use it like

```
import { NotSpecifiedCacheAdapterError } from './errors';

if (!cachedAdapter) {
   throw new NotSpecifiedCacheAdapterError();
}
```

Why? I think that by treating errors with care with custom classes for each of them will make easier for the users if they want to explicitely handle them like:

```ts
try {
  // some code
} catch (e) {
  if (e instanceof NotSpecifiedCacheAdapterError) {
     // handle this error
  }
}
```

It's something i got from python and i think that java also has this. Instead of just throwing normal Errors through the application.

Reply:
Done